gateway-service-impl/src/main/java/org/hypertrace/gateway/service/GatewayServiceImpl.java

```diff
@@ -79,6 +82,8 @@ public GatewayServiceImpl(Config appConfig) {
     QueryServiceClient queryServiceClient =
         new QueryServiceClient(new QueryServiceConfig(qsConfig));
     int qsRequestTimeout = getRequestTimeoutMillis(qsConfig);
+    ExecutorService queryExecutor =
+        QueryExecutorServiceFactory.buildExecutorService(QueryExecutorConfig.from(appConfig));
```

Comment:
are we use `ServiceFactory` suffix other places? if, not, `QueryExecutorServiceFactory` -> `QueryExecutorFactory` ?

Reply:
So I went back and forth on this - agreed we should be consistent, but the actual returned interface is an ExecutorService, so it's a factory for that, and the role of that executor service is querying.

Will proceed, but let's figure out the naming and can go back and make consistent.